pkg/extract/freebsd/freebsd_test.go

```diff
@@ -1 +1,94 @@
 package freebsd_test
+
+import (
+	"encoding/json"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/MaineK00n/vuls-data-update/pkg/extract/freebsd"
+	"github.com/MaineK00n/vuls-data-update/pkg/extract/types"
+	"github.com/MaineK00n/vuls-data-update/pkg/extract/types/detection"
+	"github.com/MaineK00n/vuls-data-update/pkg/extract/types/reference"
+)
+
+func TestExtract(t *testing.T) {
+	tests := []struct {
+		name     string
+		args     string
+		hasError bool
+	}{
+		{
+			name: "happy",
+			args: "./testdata/fixtures",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dir := t.TempDir()
+			err := freebsd.Extract(tt.args, freebsd.WithDir(dir))
+			switch {
+			case err != nil && !tt.hasError:
+				t.Error("unexpected error:", err)
+			case err == nil && tt.hasError:
+				t.Error("expected error has not occurred")
+			}
+
+			if err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+
+				if d.IsDir() {
+					return nil
+				}
+
+				file := filepath.Base(path)
+				f, err := os.Open(filepath.Join("testdata", "golden", file))
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+
+				var want types.Data
+				if err := json.NewDecoder(f).Decode(&want); err != nil {
+					return err
+				}
+
+				f, err = os.Open(path)
+				if err != nil {
+					return err
+				}
+				defer f.Close()
+
+				var got types.Data
+				if err := json.NewDecoder(f).Decode(&got); err != nil {
+					return err
+				}
+
+				opts := []cmp.Option{
+					cmpopts.SortSlices(func(a, b types.Vulnerability) bool {
+						return a.ID < b.ID
+					}),
+					cmpopts.SortSlices(func(a, b detection.Detection) bool {
+						return a.Package.Name < b.Package.Name
+					}),
+					cmpopts.SortSlices(func(a, b reference.Reference) bool {
+						return a.URL < b.URL
+					}),
+				}
+				if diff := cmp.Diff(want, got, opts...); diff != "" {
+					t.Errorf("Extract(). (-expected +got):\n%s", diff)
+				}
```

Comment:
If the output always remains the same, how about testing by comparing `[]byte`?

Reply:
Did it! https://github.com/MaineK00n/vuls-data-update/pull/174/commits/00a499584645c91f501bbfc1f8b7f6c51fa46cc2